src/OrchardCore.Modules/OrchardCore.Roles/Migrations/RolesMigrations.cs

```diff
@@ -0,0 +1,116 @@
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.BackgroundJobs;
+using OrchardCore.Data.Migration;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Scope;
+using OrchardCore.Security;
+using OrchardCore.Users;
+
+namespace OrchardCore.Roles.Migrations;
+
+public sealed class RolesMigrations : DataMigration
+{
+    private static readonly string _alternativeAdminRoleName = "SiteOwner";
+
+#pragma warning disable CA1822 // Mark members as static
+    public int Create()
+#pragma warning restore CA1822 // Mark members as static
+    {
+        ShellScope.AddDeferredTask(async scope =>
+        {
+            var roleManager = scope.ServiceProvider.GetRequiredService<RoleManager<IRole>>();
+
+            var roles = roleManager.Roles.ToList();
+
+            var adminRoles = new List<Role>();
+            var adminSystemRoleName = OrchardCoreConstants.Roles.Administrator;
+
+            foreach (var role in roles)
+            {
+                if (role is not Role r)
+                {
+                    continue;
+                }
+
+                var hasSiteOwner = r.RoleClaims.Any(x => x.ClaimValue == "SiteOwner");
```

Comment:
We need to make it work for new sites.

Reply:
It already does as during create `roles` will have no records so this logic will not be called. But for safety, I added a check against claims.